README.md

```diff
@@ -108,9 +108,11 @@ Only `host` and `username` are mandatory, but most likely you would need `databa
 - `ssl` if set to `true`, perform an attempt to connect in ssl mode, but continue unencrypted
   if encryption isn't supported by server. if set to `required` connection will fail if encryption
   is not available.
-- `ssl_opts` will be passed as is to `ssl:connect/3`
+- `ssl_opts` will be passed as is to `ssl:connect/3`. The `active` option is only available in
+  the `replication` mode and OTP >= 21.3.
 - `tcp_opts` will be passed as is to `gen_tcp:connect/3`. Some options are forbidden, such as
-  `mode`, `packet`, `header`, `active`. When `tcp_opts` is not provided, epgsql does some tuning
+  `mode`, `packet`, `header`. The `active` option is only available in the `replication` mode.
```

Comment:
I was thinking that `active` should be passed not directly to `ssl_opts`/`tcp_opts` but rather on the top-level of `epgsql:connect_opts()`, like `epgsql:connect(#{host => "..", username => "..", password => "..", socket_active => 64 | true})`. Just because all the other socket options are passed as-is to the underlying socket, but `active` is interpreted by `epgsql`. So later `epgsql_cmd_connect` and `epgsql_sock` would take the top-level `socket_active` config to set the `ssl`/`gen_tcp` `active` option based on this top-level config value.

It would also make options lookup faster, because it is just `maps:get(socket_active, ConnectOpts, true)` instead of proplist lookup in `update_active_opt` (it is important because it might be a quite hot code path when N is small)